### PR TITLE
Added TorchAO FP8 weight-only and static quantization + user guides

### PR DESCRIPTION
**Pull request by micwill755**
# Description

Adds TorchAO FP8 support to the Dynamo path: **weight-only** (`dequantize_affine` → TensorRT `IDequantizeLayer`) and **static FP8** (non-decomposed `quantize_affine_float8` / `dequantize_affine_float8`). Constant folding is updated so those DQ ops stay in the graph and are not folded into dense BF16/FP16 weights.

This matches the existing ModelOpt INT8/FP8/FP4 QDQ flow, but for TorchAO tensor-subclass checkpoints. Examples cover a toy `nn.Linear` (WOQ and static) and FLUX.1-dev WOQ. User-guide, troubleshooting, and gallery docs are updated. HTML under `docs/` is left to the post-merge docgen bot.

**Dependencies:** `torchao` (examples list it in `examples/dynamo/requirements.txt`). FP8 compile/runtime needs Hopper or newer (compute capability ≥ 9).

Replaces #4577 (same commits; reopened from `pytorch/TensorRT` branches so GitHub native Stacks work). INT4 follow-up is #4590 (2/2).

Fixes # (issue)

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by narendasan — COMMENTED** [review]
Can you add the torchao tests as a suite under full in the test suite? @lanluo-nvidia can help guide

File: docsrc/user_guide/shapes_precision/quantization.rst [hunk 1]
```diff
@@ -152,6 +155,73 @@ Quantization also works with ``torch.compile``:
...
+    model = pre_process_model_for_export(model)  # emit dequantize_affine
+
+    with exclude_dq_from_constant_folding():
+        exp_program = torch.export.export(model, (example_input,), strict=True)
+
+    trt_model = torch_tensorrt.dynamo.compile(
+        exp_program,
+        inputs=[example_input],
+        min_block_size=1,
+        use_explicit_typing=True,
```
Comment (zewenli98) [inline docsrc/user_guide/shapes_precision/quantization.rst:184]: let's remove the deprecated arg.

File: docsrc/user_guide/shapes_precision/quantization.rst [hunk 2]
```diff
@@ -152,6 +155,73 @@ Quantization also works with ``torch.compile``:
...
+``IQuantizeLayer`` / ``IDequantizeLayer``. After fusion, GEMMs can run in FP8.
+
+.. code-block:: python
+
+    quantize_static_fp8(model, (example_input,), calibration_steps=10)
+    exp_program = torch.export.export(model, (example_input,), strict=True)
+    trt_model = torch_tensorrt.dynamo.compile(
+        exp_program,
+        inputs=[example_input],
+        enabled_precisions={torch.float8_e4m3fn},
```
Comment (zewenli98) [inline docsrc/user_guide/shapes_precision/quantization.rst:215]: let's remove this as well because TRT has been strongly typed.

**Review by zewenli98 — COMMENTED** [review]
Overall LGTM. Just some deprecated arguments can be removed and need rebase.

A review from my agent (**please ignore if that's not relaed**):

quantize.py — axis derivation for FP8 quantize/dequantize
- Files: py/torch_tensorrt/dynamo/conversion/impl/quantize.py
- Issue: _fp8_scale_and_axis currently sets axis = 0 whenever scale.numel() != 1. In a symmetrical way dequantize_affine earlier derives axis from block_size (searching for a block dim == 1). The logic for FP8 quantize/dequantize should be consistent with how the per-channel scale is represented (scale shape vs block_size). Using a fixed axis=0 may be wrong for some weight layouts.
- Recommendation: derive axis based on scale.shape and/or block_size (where available), or at minimum document the assumption and add a fallback/validation that scale shape is compatible with the chosen axis. e.g., if scale is 1D with length equal to weight.shape[axis], use that axis; otherwise raise a clearer exception.

File: docsrc/user_guide/shapes_precision/quantization.rst (hunk 2 above)
Comment (micwill755) [reply docsrc/user_guide/shapes_precision/quantization.rst]: Done

File: docsrc/user_guide/shapes_precision/quantization.rst (hunk 1 above)
Comment (micwill755) [reply docsrc/user_guide/shapes_precision/quantization.rst]: Good callout on the axis derivation. The static FP8 operators do not provide block_size, so I updated _fp8_scale_and_axis to derive the axis from the input and scale shapes
